src/main/java/uk/gov/hmcts/ccd/data/casedataaccesscontrol/DefaultRoleAssignmentRepository.java

```diff
@@ -72,4 +79,49 @@ public RoleAssignmentResponse getRoleAssignments(String userId) {
             }
         }
     }
+
+    private void addETagHeader(String userId, HttpHeaders headers) {
+        if (roleAssignments.containsKey(userId)) {
+            Pair<String, RoleAssignmentResponse> stringRoleAssignmentResponsePair = roleAssignments.get(userId);
+            headers.setIfNoneMatch(stringRoleAssignmentResponsePair.getKey());
```

Comment:
do we need to set always or only when the key not matched?

Reply:
If not found in the cache we have nothing to set. It is the first invocation. Then we always set to check if it has changed.